src/io/sqlite_store.rs

```diff
@@ -0,0 +1,228 @@
+			KV_TABLE_NAME
+		);
+		let msg = format!("Failed to write to key: {}/{}", namespace, key);
+		sql_tx
+			.execute(
+				&sql,
+				named_params! {
+					":namespace": namespace,
+					":key": key,
+					":data": buf,
+				},
+			)
+			.map_err(|_| std::io::Error::new(std::io::ErrorKind::Other, msg))?;
+
+		let msg = format!("Failed to commit transaction");
+		sql_tx.commit().map_err(|_| std::io::Error::new(std::io::ErrorKind::Other, msg))?;
+
+		Ok(())
+	}
+
+	fn remove(&self, namespace: &str, key: &str) -> std::io::Result<bool> {
+		let sql = format!("DELETE FROM {} WHERE namespace=:namespace AND key=:key;", KV_TABLE_NAME);
+		let msg = format!("Failed to delete key: {}/{}", namespace, key);
+		let changes = self
+			.connection
+			.lock()
+			.unwrap()
+			.execute(
+				&sql,
+				named_params! {
+					":namespace": namespace,
+					":key": key,
+				},
+			)
+			.map_err(|_| std::io::Error::new(std::io::ErrorKind::Other, msg))?;
+
+		let was_present = changes != 0;
+
+		Ok(was_present)
+	}
+
+	fn list(&self, namespace: &str) -> std::io::Result<Vec<String>> {
+		let locked_conn = self.connection.lock().unwrap();
+
+		let sql = format!("SELECT key FROM {} WHERE namespace=:namespace", KV_TABLE_NAME);
+		let msg = format!("Failed to prepare statement");
+		let mut stmt = locked_conn
+			.prepare(&sql)
+			.map_err(|_| std::io::Error::new(std::io::ErrorKind::Other, msg))?;
+
+		let mut keys = Vec::new();
+
+		let msg = format!("Failed to retrieve queried rows");
+		let rows_iter = stmt
+			.query_map(named_params! {":namespace": namespace, }, |row| row.get(0))
+			.map_err(|_| std::io::Error::new(std::io::ErrorKind::Other, msg))?;
+
+		for k in rows_iter {
+			let msg = format!("Failed to retrieve queried rows");
+			keys.push(k.map_err(|_| std::io::Error::new(std::io::ErrorKind::Other, msg))?);
+		}
+
+		Ok(keys)
+	}
+}
+
+impl KVStorePersister for SqliteStore {
+	fn persist<W: Writeable>(&self, prefixed_key: &str, object: &W) -> lightning::io::Result<()> {
+		let msg = format!("Could not persist file for key {}.", prefixed_key);
+		let dest_file_path = PathBuf::from_str(prefixed_key).map_err(|_| {
+			lightning::io::Error::new(lightning::io::ErrorKind::InvalidInput, msg.clone())
+		})?;
+
+		let parent_directory = dest_file_path.parent().ok_or(lightning::io::Error::new(
+			lightning::io::ErrorKind::InvalidInput,
+			msg.clone(),
+		))?;
+		let namespace = parent_directory.display().to_string();
+
+		let dest_without_namespace = dest_file_path
+			.strip_prefix(&namespace)
+			.map_err(|_| lightning::io::Error::new(lightning::io::ErrorKind::InvalidInput, msg))?;
+		let key = dest_without_namespace.display().to_string();
+
+		self.write(&namespace, &key, &object.encode())?;
+		Ok(())
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::test::utils::random_storage_path;
+	use lightning::util::persist::KVStorePersister;
+	use lightning::util::ser::Readable;
+
+	use proptest::prelude::*;
+	proptest! {
```

Comment:
Perhaps I'm misunderstanding, but what is the value of using `proptest` here. The data written doesn't seem relevant to exercising the code paths.